hotel_reservation/models/hotel_reservation.py

```diff
@@ -279,45 +287,91 @@ def confirmed_reservation(self):
-                             from hotel_reservation as hr "
-                             "inner join hotel_reservation_line as \
-                             hrl on hrl.line_id = hr.id "
-                             "inner join hotel_reservation_line_room_rel \
-                             as hrlrr on hrlrr.room_id = hrl.id "
-                             "where hr.id = cast(%s as integer) )",
-                             (reservation.checkin, reservation.checkout,
-                              str(reservation.id), str(reservation.id)))
-            res = self._cr.fetchone()
-            roomcount = res and res[0] or 0.0
-            if roomcount:
-                raise except_orm(_('Warning'), _('You tried to confirm \
-                reservation with room those already reserved in this \
-                reservation period'))
-            else:
-                self.state = 'confirm'
-                for line_id in reservation.reservation_line:
-                    line_id = line_id.reserve
-                    for room_id in line_id:
-                        vals = {
-                            'room_id': room_id.id,
-                            'check_in': reservation.checkin,
-                            'check_out': reservation.checkout,
-                            'state': 'assigned',
-                            'reservation_id': reservation.id,
-                            }
-                        room_id.write({'isroom': False, 'status': 'occupied'})
-                        reservation_line_obj.create(vals)
+            for line_id in reservation.reservation_line:
+                for room_id in line_id.reserve:
+                    if room_id.room_reservation_line_ids:
+                        for reserv in room_id.room_reservation_line_ids.\
+                                search([('status', '=', 'confirm')]):
+                            reserv_checkin = dtime.\
+                                strptime(reservation.checkin, dt)
+                            reserv_checkout = dtime.\
+                                strptime(reservation.checkout, dt)
+                            check_in = dtime.strptime(reserv.check_in, dt)
+                            check_out = dtime.strptime(reserv.check_out, dt)
+                            range1 = [reserv_checkin, reserv_checkout]
+                            range2 = [check_in, check_out]
+                            overlap_dates = self.check_overlap(*range1) \
+                                & self.check_overlap(*range2)
+                            if overlap_dates:
+                                raise ValidationError(_('You tried to Confirm '
+                                                        'reservation with room'
+                                                        ' those already '
+                                                        'reserved in this '
+                                                        'Reservation Period'))
```

Comment:
@krishnaprajapati16894 we have overlap dates, we may add in validation error that those dates are overlaps.